src/main/java/org/opensearch/knn/index/vectorvalues/KNNFloatVectorValues.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.vectorvalues;
+
+import org.apache.lucene.codecs.KnnFieldVectorsWriter;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.FloatVectorValues;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.knn.index.codec.util.KNNVectorSerializer;
+import org.opensearch.knn.index.codec.util.KNNVectorSerializerFactory;
+
+import java.io.IOException;
+
+/**
+ * Concrete implementation of {@link KNNVectorValues} that returns float[] as vector and provides an abstraction over
+ * {@link BinaryDocValues}, {@link FloatVectorValues}, {@link KnnFieldVectorsWriter} etc.
+ */
+public class KNNFloatVectorValues extends KNNVectorValues<float[]> {
+    KNNFloatVectorValues(final KNNVectorValuesIterator vectorValuesIterator) {
+        super(vectorValuesIterator);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public float[] getVector() throws IOException {
+        final float[] vector;
+        if (vectorValuesIterator instanceof KNNVectorValuesIterator.DocIdsIteratorValues) {
+            final DocIdSetIterator docIdSetIterator = vectorValuesIterator.getDocIdSetIterator();
+            if (docIdSetIterator instanceof BinaryDocValues) {
+                final BinaryDocValues values = (BinaryDocValues) docIdSetIterator;
+                vector = getFloatVectorFromByteRef(values.binaryValue());
+                dimension = vector.length;
```

Comment:
Please apply DRY principle for this part of code.

Reply:
Can we think of segregation these if elses to different methods which represents the functionality. This way we can reduce clumsiness in this method and achieve better readabitlity.

Reply:
actually if you look closely it looks similar but they are not. Nevertheless the code has been refactored now, which should solve your concern too.